### DISCUSSION TITLE
Interactive between ghelper and Nvidia App One Click Performance Tuning

### DISCUSSION THREAD

**Discussion by umumpaul** [seerge/g-helper#2686] [General]
I don't know if Nvidia App will overwrite the setting of g-helper, or vice versa, or they real the GPU config from same place under the hood.

detail here:

https://www.nvidia.com/en-us/geforce/news/nvidia-app-beta-update-av1-performance-tuning/

**Comment by seerge** [comment]
@umumpaul hello, I don't think that nvidia's app would offer tuning for the laptops :) Most probably only for desktop GPUs

But anyway, the only thing that G-Helper can set for Nvidia - is Core and Memory Clock offsets. It does it via Nvidia's own "official" API (same as Afterburner for example).

**Reply by umumpaul** [reply]
the link state that

Owners of GeForce RTX desktop graphics cards and **laptops** can monitor and maximize the potential of their GPUs with the Performance Panel and one-click automatic GPU tuning.

If g-helper read and write the adjustment via official Nvidia API, I suppose the result of the one-click automatic HPU tuning will reflect on g-helper too?

**Reply by seerge** [reply]
@umumpaul if their app adjusts same 2 values - may be :) I don't know yet.

But currently Fans + Power reads and shows current overclock settings under GPU tab (unless custom ones are already applied in G-Helper itself of course)

**Comment by ricoded-o00** [comment]
As it happens at the moment, for me at least, Nvidia App's performance tuning overwrote G-Helper settings for my RTX3080 Laptop (in a 2021 ROG STRIX 17) but it wasn't reflected in G-Helper. Changing a profile in G-Helper after Nvidia App did it's thing (or a restart) results in a Windows UAC prompt and on confirming, G-Helper will apply it's own profile.

As for the "performance tunning", it's extremely conservative while also not considering undervolting in addition to the GPU staying at 85-86C. The actual numbers for my machine was a whole +41MHz increase on the Core Clock and 7001MHz VRAM clock ceiling, which is more-or less the AC defaults if I'm not mistaking, Performance was not stellar either, I got far better results limiting the Core Clock and increasing the Offset manually and VRAM ceiling to 7301MHz (as far as it will go for me) and I can run that all day with the fans screaming at 76-82C.

Again, my experience on my laptop. Your mileage may vary.

**Reply by seerge** [reply]
@ricoded-o00 if you would open `Fans + Power` tab without changing mode in the app, it re-reads Clock Offset settings from the driver (and displays them). Do they reflect anything done by Nvidia app like that ? How do you see that it's +41MHz increase?

**Reply by ricoded-o00** [reply]
> @ricoded-o00 if you would open `Fans + Power` tab without changing mode in the app, it re-reads Clock Offset settings from the driver (and displays them). Do they reflect anything done by Nvidia app like that ? How do you see that it's +41MHz increase?

I'm running custom profiles so that might have to do with it so I probably should have mentioned it. I don't really have time to play with that today but I'll try to recreate everything when I get some time in a day or two since I disabled the whole thing.

+41MHz was reported by NVidia App (might have had to specify that too). I drew the conclusion from the reported clocks from Nvidia's Overlay, specifically the Memory Clocks jumping between 6001MHz and 7001MHz instead of staying at 7301MHz (as per +300 specified by me in G-Helper) which changed only after applying another mode (Turbo if I remember correctly) and then going back to my Custom one. Also confirmed by a significantly lower benchmark score. I don't have any screenshots though.

As for G-Helper's `Fans + Power` (opened on my custom profile) didn't show any change, it was still showing the numbers I set while Nvidia Overlay was showing the actual numbers. When I activated the Turbo profile, I got a UAC prompt to allow it and then the clocks changed. Speaking of those display updates in G-Helper - changing between the custom profiles from the UI doesn't update the numbers on the tiles until I click the sliders, then they update to what that profile has, similar to my mouse tile update problem [here](https://github.com/seerge/g-helper/issues/2692#issue-2335154589).

Now thinking of what I'm typing and the way the Nvidia App gradually increases parameters, it probably does the same as GeForce Experience did and overrides everything until something else is applied by G-Helper. Which basically is the same thing AC's ROG Boost does, it would correspond to my 1540-ish MHz in AC.

**Reply by seerge** [reply]
@ricoded-o00 it could be that NVidia app does it differently, or adjusts on the fly - after all they interact with their own drivers and there is no way to check what and how exactly it does that.

G-Helper , Armoury Crate (in Manual mode) and Afterbuner use NVidia API :) So it's up to you where to overclock your GPU. Results should be the same tho (assuming same values at least)

**Reply by ricoded-o00** [reply]
True! I was just posting my experience/observations with it. I personally do prefer G-Helper since it's letting me do a lot in one small application instead of chasing settings in multiple applications.
